### PR TITLE
Feat/dynamite/octet stream

### PR DESCRIPTION
**Pull request by Leptopoda**
pulled from #764

### CODE REVIEW

**Review by provokateurin — CHANGES_REQUESTED** [review]
This PR has 3 different changes and all of them are not related to each other. Please split them into 3 PRs

**Comment by Leptopoda** [comment]
It's currently based on #846 and I'll split it up (do separate PRs) once it is merged

**Comment by provokateurin** [comment]
I guess it's ready for review?

File: packages/dynamite/dynamite/lib/src/models/type_result/type_result.dart [hunk 1]
```diff
@@ -87,6 +87,11 @@ sealed class TypeResult {
         return 'json.encode($serialized)';
       case 'application/x-www-form-urlencoded':
         return 'Uri(queryParameters: $serialized! as Map<String, dynamic>).query';
+      case 'application/octet-stream':
+        if (className != 'Uint8List') {
+          throw Exception('octet-stream can only be applied to binary data. Expected Uint8List but got $className');
+        }
```
Comment (provokateurin) [inline packages/dynamite/dynamite/lib/src/models/type_result/type_result.dart:93]: Is this check actually necessary?

Comment (Leptopoda) [reply packages/dynamite/dynamite/lib/src/models/type_result/type_result.dart:93]: This method is a generic implementation for all TypeResults (unless overridden).

It's just some safety mechanism.

**Pull request merged**